ocaml/src/proxy_server.ml

```diff
@@ -60,20 +58,30 @@ let albamgr_cfg_to_ini_string (cluster_id, nodes) =
 
   ini_hash_to_string h
 
-let write_albamgr_cfg albamgr_cfg destination =
-  let s = albamgr_cfg_to_ini_string albamgr_cfg in
-  let tmp = destination ^ ".tmp" in
-  Lwt_extra2.unlink ~may_not_exist:true tmp >>= fun () ->
-  Lwt_extra2.with_fd
-    tmp
-    ~flags:Lwt_unix.([ O_WRONLY; O_CREAT; O_EXCL; ])
-    ~perm:0o664
-    (fun fd ->
-       Lwt_extra2.write_all
-         fd
-         s 0 (String.length s) >>= fun () ->
-       Lwt_unix.fsync fd) >>= fun () ->
-  Lwt_unix.rename tmp destination
+let maybe_write_albamgr_cfg albamgr_cfg = function
+  | Url.File destination ->
+     let s = albamgr_cfg_to_ini_string albamgr_cfg in
+     let tmp = destination ^ ".tmp" in
+     Lwt.catch
+       (fun () ->
+        Lwt_extra2.unlink ~may_not_exist:true tmp >>= fun () ->
+        Lwt_extra2.with_fd
+          tmp
+          ~flags:Lwt_unix.([ O_WRONLY; O_CREAT; O_EXCL; ])
+          ~perm:0o664
+          (fun fd ->
+           Lwt_extra2.write_all
+             fd
+             s 0 (String.length s) >>= fun () ->
+           Lwt_unix.fsync fd) >>= fun () ->
+        Lwt_unix.rename tmp destination
+       )
+       (fun exn ->
+        Lwt_log.info_f ~exn
+                       "couldn't write config to destination:%s" destination
+       )
+  | (Url.Etcd etcd) as destination ->
+     Lwt_log.info_f "refresh NOT pushing to %s" (Url.show destination)
```

Comment:
The proxy has it's own key for storing the abm configuration, so I'd expect that the proxy keeps on updating its key like it does for when it's a file instead of an etcd key. It has it's own duplicated key because the code was - like @domsj stated - designed to be able to run a proxy on another cluster.